File: backend/internal/vo/agent.go
package vo

import "time"

type CourseAgentVO struct {
	ID             uint64    `json:"id"`
	CourseID       uint64    `json:"courseId"`
	AgentName      string    `json:"agentName"`
	PromptTemplate string    `json:"promptTemplate,omitempty"`
	Status         string    `json:"status"`
	RetrievalScope string    `json:"retrievalScope"`
	CreatedBy      uint64    `json:"createdBy"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

type AgentConversationVO struct {
	ID                uint64    `json:"id"`
	CourseID          uint64    `json:"courseId"`
	AgentID           uint64    `json:"agentId"`
	UserID            uint64    `json:"userId"`
	ConversationTitle string    `json:"conversationTitle"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
}

type AgentMessageSourceVO struct {
	ID             uint64    `json:"id"`
	MessageID      uint64    `json:"messageId"`
	MaterialNodeID uint64    `json:"materialNodeId"`
	FileName       string    `json:"fileName,omitempty"`
	SnippetText    string    `json:"snippetText"`
	CreatedAt      time.Time `json:"createdAt"`
}

type AgentMessageVO struct {
	ID             uint64                 `json:"id"`
	ConversationID uint64                 `json:"conversationId"`
	SenderType     string                 `json:"senderType"`
	MessageContent string                 `json:"messageContent"`
	TokenUsage     int                    `json:"tokenUsage"`
	CreatedAt      time.Time              `json:"createdAt"`
	Sources        []AgentMessageSourceVO `json:"sources,omitempty"`
}

type AgentConversationDetailVO struct {
	Conversation AgentConversationVO `json:"conversation"`
	Messages     []AgentMessageVO    `json:"messages"`
}

type AgentAskResultVO struct {
	ConversationID uint64                 `json:"conversationId"`
	Question       string                 `json:"question"`
	Answer         string                 `json:"answer"`
	Sources        []AgentMessageSourceVO `json:"sources"`
}

File: frontend/src/types/agent.ts
export type AgentStatus = 'enabled' | 'disabled'
export type AgentRetrievalScope = 'course_all'
export type AgentSenderType = 'user' | 'agent'

export interface CourseAgentVO {
  id: number
  courseId: number
  agentName: string
  promptTemplate?: string
  status: AgentStatus
  retrievalScope: AgentRetrievalScope
  createdBy: number
  createdAt: string
  updatedAt: string
}

export interface UpdateCourseAgentPayload {
  agentName: string
  promptTemplate: string
  status: AgentStatus
  retrievalScope: AgentRetrievalScope
}

export interface AgentConversationVO {
  id: number
  courseId: number
  agentId: number
  userId: number
  conversationTitle: string
  createdAt: string
  updatedAt: string
}

export interface CreateAgentConversationPayload {
  title: string
}

export interface AgentMessageSourceVO {
  id?: number
  messageId?: number
  materialNodeId: number
  fileName?: string
  snippetText: string
  createdAt?: string
}

export interface AgentMessageVO {
  id: number
  conversationId: number
  senderType: AgentSenderType
  messageContent: string
  tokenUsage: number
  createdAt: string
  sources?: AgentMessageSourceVO[]
}

export interface AgentConversationDetailVO {
  conversation: AgentConversationVO
  messages: AgentMessageVO[]
}

export interface AskAgentPayload {
  conversationId: number
  question: string
}

export interface AgentAskResultVO {
  conversationId: number
  question: string
  answer: string
  sources: AgentMessageSourceVO[]
}

export interface CourseAgentStreamDeltaEvent {
  content: string
}

export interface CourseAgentStreamCompleteEvent {
  answer: string
  sources: AgentMessageSourceVO[]
  tokenUsage: number
}
